File: CS/ReplicateBuiltInToolbar/ViewModels/EmbedImageSettingsViewModel.cs
using DevExpress.Maui.Core;
using DevExpress.Maui.HtmlEditor;
using Microsoft.Maui.Controls;

namespace HtmlEditToolbarCustomization;

public class EmbedImageSettingsViewModel : BaseViewModel {
    public EmbedImageSettingsViewModel(HtmlEdit owner, Action closeDelegate) : base(owner) {
        CloseDelegate = closeDelegate;
    }

    public Command ShowImagePickerCommand => new Command(ShowImagePicker);
    public Command ShowPhotoPickerCommand => new Command(ShowPhotoPicker);
    Action CloseDelegate { get; }

    async void ShowPhotoPicker(object obj) {
        if (Owner == null)
            return;
        FileResult result = await MediaPicker.CapturePhotoAsync();
        if (result == null)
            return;
        await Owner.InsertImageAsync(ImageSource.FromFile(result.FullPath), new HtmlSize(100, HtmlSizeUnit.Percent), new HtmlSize(100, HtmlSizeUnit.Percent));
        CloseDelegate?.Invoke();
    }
    async void ShowImagePicker(object obj) {
        if (Owner == null)
            return;
        FileResult result = await MediaPicker.PickPhotoAsync();
        if (result == null)
            return;
        await Owner.InsertImageAsync(ImageSource.FromFile(result.FullPath), new HtmlSize(100, HtmlSizeUnit.Percent), new HtmlSize(100, HtmlSizeUnit.Percent));
        CloseDelegate?.Invoke();
    }
}

File: VB/ReplicateBuiltInToolbar/ViewModels/EmbedImageSettingsViewModel.vb
Imports DevExpress.Maui.Core
Imports DevExpress.Maui.HtmlEditor
Imports Microsoft.Maui.Controls

Namespace HtmlEditToolbarCustomization

    Public Class EmbedImageSettingsViewModel
        Inherits BaseViewModel

        Public Sub New(ByVal owner As HtmlEdit, ByVal closeDelegate As Action)
            MyBase.New(owner)
            Me.CloseDelegate = closeDelegate
        End Sub

        Public ReadOnly Property ShowImagePickerCommand As Command
            Get
                Return New Command(AddressOf ShowImagePicker)
            End Get
        End Property

        Public ReadOnly Property ShowPhotoPickerCommand As Command
            Get
                Return New Command(AddressOf ShowPhotoPicker)
            End Get
        End Property

        Private ReadOnly Property CloseDelegate As Action

        Private Async Sub ShowPhotoPicker(ByVal obj As Object)
            If Owner Is Nothing Then Return
            Dim result As FileResult = Await MediaPicker.CapturePhotoAsync()
            If result Is Nothing Then Return
            Await Owner.InsertImageAsync(ImageSource.FromFile(result.FullPath), New HtmlSize(100, HtmlSizeUnit.Percent), New HtmlSize(100, HtmlSizeUnit.Percent))
            CloseDelegate?.Invoke()
        End Sub

        Private Async Sub ShowImagePicker(ByVal obj As Object)
            If Owner Is Nothing Then Return
            Dim result As FileResult = Await MediaPicker.PickPhotoAsync()
            If result Is Nothing Then Return
            Await Owner.InsertImageAsync(ImageSource.FromFile(result.FullPath), New HtmlSize(100, HtmlSizeUnit.Percent), New HtmlSize(100, HtmlSizeUnit.Percent))
            CloseDelegate?.Invoke()
        End Sub
    End Class
End Namespace
